### DISCUSSION TITLE
How can I know if a column / cell / header is a display column / cell / header?

### DISCUSSION THREAD

**Discussion by jonahallibone** [TanStack/table#5223] [General]
I have a few tables with columns that are buttons, checkboxes, etc. I have a few functions where I export all the "data" from the table, but I'd like to be able to filter out columns / headers which are explicitly created via `columnHelper.display(..)`

I don't see anything obvious on the cell / header that would make me think there is some dynamic way of filtering these items. Should I make a PR for this feature, or is this something outside the purview of the library?

**Comment by harisrahman** [comment]
\+ 1

**Comment by julienmiclo** [comment]
Hello,

if you use  `enableHiding: false` in your column definition, you can do this :

```
table
.getAllColumns()
.filter((column) => typeof column.accessorFn !== "undefined" && column.getCanHide())
.map((column) => console.log(column))
```
or you can define meta.

Regards,
